bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/handler/EtherRainHandler.java

```diff
@@ -0,0 +1,240 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EtherRainHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Joe Inkenbrandt - Initial contribution
+ */
+public class EtherRainHandler extends BaseThingHandler {
+
+  private final Logger logger = LoggerFactory.getLogger(EtherRainHandler.class);
+
+  private EtherRainCommunication device = null;
+  private boolean connected = false;
+
+  @Nullable
+  private ScheduledFuture<?> updateJob;
+
+  /*
+   * Constructor class. Only call the parent constructor
+   */
+  public EtherRainHandler(Thing thing) {
+    super(thing);
+  }
+
+  @Override
+  public void handleCommand(ChannelUID channelUID, Command command) {
+    if (command.toString().equals("REFRESH")) {
+      updateBridge();
+      return;
+    }
+
+    else if (channelUID.getId().equals(EtherRainBindingConstants.CHANNEL_ID_EXECUTE)) {
+      execute();
+      updateState(EtherRainBindingConstants.CHANNEL_ID_EXECUTE, OnOffType.OFF);
+    }
+
+    else if (channelUID.getId().equals(EtherRainBindingConstants.CHANNEL_ID_CLEAR)) {
+      clear();
+      updateState(EtherRainBindingConstants.CHANNEL_ID_CLEAR, OnOffType.OFF);
+    } else {
+      logger.warn("Unhandled command: " + command.toFullString());
+    }
+
+  }
+
+  private boolean connectBridge() {
+    EtherRainConfiguration config = getConfigAs(EtherRainConfiguration.class);
+
+    logger.debug("Attempting to connect to Etherrain wtih config = (Host: {}, Port: {}, Password: {}, Refresh: {}).",
+        config.host, config.port, config.password, config.refresh);
+
+    device = new EtherRainCommunication(config.host, config.port, config.password);
+
+    EtherRainStatusResponse response;
+    try {
+      response = device.commandStatus();
+    } catch (EtherRainException | IOException e) {
+      updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+          "Could not create a connection to the EtherRain");
+      logger.debug("Could not open API connection to the EtherRain device. Exception received: {}", e.toString());
+      device = null;
+      updateStatus(ThingStatus.OFFLINE);
+      return false;
+    }
+    if (response == null) {
+      logger.debug("Command Status returned null");
+      device = null;
+      updateStatus(ThingStatus.OFFLINE);
+      return false;
+    }
+
+    updateStatus(ThingStatus.ONLINE);
+
+    return true;
+  }
+
+  private void startUpdateJob() {
+    logger.debug("Starting Etherrain Update Job");
+    updateJob = scheduler.scheduleWithFixedDelay(this::updateBridge, 0,
+        getConfigAs(EtherRainConfiguration.class).refresh, TimeUnit.SECONDS);
+
+    logger.debug("EtherRain sucessfully initialized. Starting status poll at: "
+        + getConfigAs(EtherRainConfiguration.class).refresh);
+
+  }
+
+  private void stopUpdateJob() {
+    logger.debug("Stopping Etherrain Update Job");
+    if (updateJob != null && !updateJob.isDone()) {
```

Comment:
Assign to a local (final) Variable, this prevents NPEs

Reply:
Again, finally had time to research this and I am a bit confused. It's been frustrating because I have been working on this as the development environment has been in flux. At some point, the IDE required me to put the @Nullable flag on my updateJob variable. After reading your comment, I went to the Openhab documentation which uses the Wemo binding as an example on how to start a thread. I use the exact same logic as that binding (without a final variable). The only difference......no @Nullable on the thread variable. So I removed that from my code to match the example given. If you would like me to use a different pattern, please provide an example.